src/features/vault.ts

```diff
@@ -0,0 +1,247 @@
+      vscode.window.showInformationMessage(`File decrypted: '${doc.fileName}'`);
+    }
+    vscode.commands.executeCommand('workbench.action.files.revert');
+  }
+};
+
+// Returns whether the selected text is encrypted or in plain text.
+const getInlineTextType = (text: string) => {
+  if (text.trim().startsWith('!vault |')) {
+    text = text.replace('!vault |', '');
+  }
+
+  return text.trim().startsWith('$ANSIBLE_VAULT;') ? 'encrypted' : 'plaintext';
+};
+
+// Returns wheter the file is encrypted or in plain text.
+const getTextType = (text: string) => {
+  return text.indexOf('$ANSIBLE_VAULT;') === 0 ? 'encrypted' : 'plaintext';
+};
+
+const encryptInline = (
+  text: string,
+  rootPath: string | undefined,
+  vaultId: string,
+  config: vscode.WorkspaceConfiguration
+) => {
+  const tmpFilename = tmp.tmpNameSync();
+  fs.writeFileSync(tmpFilename, Buffer.from(text, 'utf8'));
+  console.log(`Wrote encrypted string to temporary file '${tmpFilename}'`);
+
+  encryptFile(tmpFilename, rootPath, vaultId, config);
+  const encryptedText = fs.readFileSync(tmpFilename, 'utf8');
+  console.log(`encryptedText == '${encryptedText}'`);
+
+  if (!!tmpFilename) {
+    fs.unlinkSync(tmpFilename);
+    console.log(`Removed temporary file: '${tmpFilename}'`);
+  }
+
+  return encryptedText.trim();
+};
+
+const decryptInline = (
+  text: string,
+  rootPath: string | undefined,
+  config: vscode.WorkspaceConfiguration
+) => {
+  // Delete inline vault prefix, then trim spaces and newline from the entire string and, at last, trim the spaces in the multiline string.
+  text = text
+    .replace('!vault |', '')
+    .trim()
+    .replace(/[^\S\r\n]+/gm, '');
+
+  const tmpFilename = tmp.tmpNameSync();
+  fs.writeFileSync(tmpFilename, Buffer.from(text, 'utf8'));
+  console.log(`Wrote encrypted string to temporary file '${tmpFilename}'`);
+
+  decryptFile(tmpFilename, rootPath, config);
+  const decryptedText = fs.readFileSync(tmpFilename, 'utf8');
+  console.log(`decryptedText == '${decryptedText}'`);
+
+  if (!!tmpFilename) {
+    fs.unlinkSync(tmpFilename);
+    console.log(`Removed temporary file: '${tmpFilename}'`);
+  }
+
+  return decryptedText;
+};
+
+const encryptFile = (
+  f: string,
+  rootPath: string | undefined,
+  vaultId: string,
+  config: vscode.WorkspaceConfiguration
+) => {
+  console.log(`Encrypt file: ${f}`);
+
+  let cmd = `${config.executablePath} encrypt "${f}"`;
+  cmd += ` --encrypt-vault-id="${vaultId}"`;
+
+  if (!!rootPath) {
+    exec(cmd, { cwd: rootPath });
+  } else {
+    exec(cmd);
+  }
+};
+
+const decryptFile = (
+  f: string,
+  rootPath: string | undefined,
+  config: vscode.WorkspaceConfiguration
+) => {
+  console.log(`Decrypt file: ${f}`);
+
+  const cmd = `${config.executablePath} decrypt "${f}"`;
+
+  if (!!rootPath) {
+    exec(cmd, { cwd: rootPath });
+  } else {
+    exec(cmd);
+  }
+};
+
+const exec = (cmd: string, opt = {}) => {
+  console.log(`> ${cmd}`);
+  return cp.execSync(cmd, opt);
```

Comment:
I don't like that there's a lot of blocking invocations that can potentially be disruptive from the UX perspective and annoy the end-users by blocking the rendering. Any chance to change all of this to async?